backends/healthchecks/healthchecks.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthchecks
```

Comment:
wait im confused, why is it a whole backend ? maybe adda. comment here

Reply:
well, it's a folder under `backends` yes lol should we put a README.md in the `backends` to specicify it contains backends? :)

Reply:
(rereading is better) the healthchecks handled here are specific to LBs scenarios and give information about internal names to the cloud provider. So it's (1) something not everyone wants and (2) totally orthogonal to any backend managing connectivity. The same orthogonality will probably appear again when I'll take the time to code a node-local DNS service ;)